src/lib/ccf/index.ts

```diff
@@ -0,0 +1,458 @@
+  private awsList: KeyValuePair = {};
+
+  private provider = '';
+  private instanceType = '';
+  private expectedLifespan = 4;
+
+  private interpolation = Interpolation.LINEAR;
+
+  constructor() {
+    this.standardizeInstanceMetrics();
+  }
+
+  /**
+   * Defined for compatibility. Not used in CCF.
+   */
+  authenticate(authParams: object): void {
+    this.authParams = authParams;
+  }
+
+  /**
+   *  Configures the CCF Plugin for IEF
+   *  @param {string} name name of the resource
+   *  @param {Object} staticParams static parameters for the resource
+   *  @param {("aws"|"gcp"|"azure")} staticParams.provider aws, gcp, azure
+   *  @param {string} staticParams.instance_type instance type from the list of supported instances
+   *  @param {number} staticParams.expected_lifespan expected lifespan of the instance in years
+   *  @param {Interpolation} staticParams.interpolation linear(All Clouds), spline (only for AWS)
+   */
+  async configure(
+    name: string,
+    staticParams: object | undefined = undefined
+  ): Promise<IImpactModelInterface> {
+    this.name = name;
+
+    if (staticParams === undefined) {
+      throw new Error('Required Parameters not provided');
+    }
+
+    if ('provider' in staticParams) {
+      const provider = staticParams?.provider as string;
+      if (['aws', 'gcp', 'azure'].includes(provider)) {
+        this.provider = provider;
+      } else {
+        throw new Error('Provider not supported');
+      }
+    } else {
+      throw new Error('Provider not provided');
+    }
+
+    if ('instance_type' in staticParams) {
+      const instanceType = staticParams?.instance_type as string;
+      if (instanceType in this.computeInstances[this.provider]) {
+        this.instanceType = instanceType;
+      } else {
+        throw new Error('Instance Type not supported');
+      }
+    } else {
+      throw new Error('Instance Type not provided');
+    }
+
+    if ('expected_lifespan' in staticParams) {
+      this.expectedLifespan = staticParams?.expected_lifespan as number;
+    }
+
+    if ('interpolation' in staticParams) {
+      if (this.provider !== 'aws') {
+        throw new Error('Interpolation method not supported');
+      }
+      const interpolation = staticParams?.interpolation as Interpolation;
+      if (Object.values(Interpolation).includes(interpolation)) {
+        this.interpolation = interpolation;
+      } else {
+        throw new Error('Interpolation method not supported');
+      }
+    }
+
+    return this;
+  }
+
+  /**
+   * Calculate the total emissions for a list of observations
+   *
+   * Each Observation require:
+   *  @param {Object[]} observations  ISO 8601 datetime string
+   *  @param {string} observations[].datetime ISO 8601 datetime string
+   *  @param {number} observations[].duration observation duration in seconds
+   *  @param {number} observations[].cpu percentage cpu usage
+   */
+  async calculate(
+    observations: object | object[] | undefined
```

Comment:
if we know what kind of object is observation, then let's use that definition instead of object

Reply:
@narekhovhannisyan it is kept that way to avoid issues when compiling to different languages in JSII